java-checks/src/main/java/org/sonar/java/checks/StaticMethodCheck.java

```diff
@@ -48,7 +48,10 @@
     MethodMatcher.create()
       .typeDefinition(TypeCriteria.subtypeOf(JAVA_IO_SERIALIZABLE)).name("writeObject").addParameter(TypeCriteria.subtypeOf("java.io.ObjectOutputStream")),
     MethodMatcher.create()
-      .typeDefinition(TypeCriteria.subtypeOf(JAVA_IO_SERIALIZABLE)).name("readObjectNoData").withoutParameter());
+      .typeDefinition(TypeCriteria.subtypeOf(JAVA_IO_SERIALIZABLE)).name("readObjectNoData").withoutParameter(),
+    MethodMatcher.create()
+      .typeDefinition(TypeCriteria.subtypeOf(JAVA_IO_SERIALIZABLE)).name("readResolve").withoutParameter()
```

Comment:
According to Serializable javadoc, maybe we should add `writeReplace()` as well (https://docs.oracle.com/javase/10/docs/api/java/io/Serializable.html). But this really needs a proper ticket and PR, and should not be hidden in that change.